Objective-C/Internal/Replicator/CBLWebSocket.mm

```diff
@@ -483,7 +474,17 @@ - (void) receivedHTTPResponse: (CFHTTPMessageRef)httpResponse {
         NSMutableArray* cookies = [NSMutableArray arrayWithCapacity: matches.count];
         for (NSTextCheckingResult* match in matches) {
             if ([match range].length > 0) {
-                [cookies addObject: [cookie substringWithRange: [match range]]];
+                NSString* cookieStr = [cookie substringWithRange: [match range]];
+                [cookies addObject: cookieStr];
+                
+                // save to lite core
+                C4Error err = {};
+                CBLStringBytes cookieSlice(cookieStr);
+                CBLStringBytes host(_remoteURL.host);
+                CBLStringBytes path(_remoteURL.path.stringByDeletingLastPathComponent);
+                if (!c4db_setCookie(_c4db, cookieSlice, host, path, &err)) {
```

Comment:
Can you confirm here too if the db needs to be locked when calling the function?